compiler/src/scala/quoted/runtime/impl/QuoteMatcher.scala

```diff
@@ -341,19 +383,22 @@ object QuoteMatcher {
             case scrutinee @ DefDef(_, paramss1, tpt1, _) =>
               pattern match
                 case pattern @ DefDef(_, paramss2, tpt2, _) =>
-                  def rhsEnv: Env =
-                    val paramSyms: List[(Symbol, Symbol)] =
-                      for
-                        (clause1, clause2) <- paramss1.zip(paramss2)
-                        (param1, param2) <- clause1.zip(clause2)
-                      yield
-                        param1.symbol -> param2.symbol
-                    val oldEnv: Env = summon[Env]
-                    val newEnv: List[(Symbol, Symbol)] = (scrutinee.symbol -> pattern.symbol) :: paramSyms
-                    oldEnv ++ newEnv
-                  matchLists(paramss1, paramss2)(_ =?= _)
-                    &&& tpt1 =?= tpt2
-                    &&& withEnv(rhsEnv)(scrutinee.rhs =?= pattern.rhs)
+                  def matchParamss(scparamss: List[ParamClause], ptparamss: List[ParamClause])(using Env): optional[(Env, MatchingExprs)] =
```

Comment:
This method was necessary to match with nested dependent types such as
```
case '{
      def p(dsl: DSL)(a: dsl.N): dsl.N = a
      $y(p): String
    } =>
      '{ $y((dsl: DSL) => (b2: dsl.N) => dsl.next(b2)) }
```
where second param types are dependent on first params.

Reply:
Yes, I've added it to test cases. https://github.com/lampepfl/dotty/pull/17567/files#diff-f7f4e796d4ad68562b3441959fa979ec8c04b2ff8981ae8a9bf7c7f83555d389R12-R19